include/boost/json/detail/impl/number.ipp

```diff
@@ -348,10 +391,18 @@ loop:
                     ++p;
                     pos_ = dig_;
                     n_.kind = kind::double_;
-                    continue;
+                    n_.u = m;
+                    st_ = state::mant_frac_1_digit;
+                    goto loop;
                 }
                 if(*p == 'e' || *p == 'E')
                 {
+                    // if we see an exponent but have not yet seen a decimal point
+                    // then the decimal point must be assumed to be at the position of
+                    // the [eE]
+                    // e.g. 100e0 is equivalent to 100.0
```

Comment:
LOL what is this wall of text? No one wants to read that, and no one will read that... consider `// 'E' counts as '.'` instead

Reply:
I'd want to read it :)

Resolved with `// 'E' implies '.' if not already encountered`